lib/wanda/messaging/mapper.ex

```diff
@@ -15,4 +17,23 @@ defmodule Wanda.Messaging.Mapper do
       {:ok, cloud_event}
     end
   end
+
+  @spec to_json(struct()) :: binary() | {:error, any()}
+  def to_json(event) do
+    with {:ok, type, identifier} <- EventsRegistry.extract_metadata(event),
+         {:ok, cloud_event} <- build_cloud_event(identifier, type, event) do
+      Cloudevents.to_json(cloud_event)
+    end
+  end
+
+  defp build_cloud_event(id, type, event) do
+    %{
+      "specversion" => "1.0",
+      "id" => id,
+      "type" => type,
+      "source" => "https://github.com/trento-project/wanda",
+      "data" => event
+    }
+    |> CloudEvent.from_map()
+  end
```

Comment:
My doubt is that we end up doing the exact same work in the contract repo, are we sure we wanna move forward with this?